### PR TITLE
feat(RHINENG-14055): Ability to filter Reports by percent_compliant

### PR DESCRIPTION
**Pull request by LightOfHeaven1994**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/models/v2/report.rb [hunk 1]
```diff
@@ -114,6 +114,12 @@ class Report < ApplicationRecord
 
       { conditions: "v2_policies.id IN (#{ids.to_sql})" }
     end
+    searchable_by :percent_compliant, %i[eq gt lt gte lte], except_parents: %i[systems] do |_key, op, val|
```
Comment (skateman) [inline app/models/v2/report.rb:117]: You should cover all cases with tests

Comment (LightOfHeaven1994) [reply app/models/v2/report.rb:117]: @skateman done

File: spec/fixtures/files/searchable/reports_controller.yaml [hunk 2]
```diff
@@ -150,3 +150,118 @@
...
+        :os_major_version: 8
+        :system_id: ${system_id}
+        :compliant_system_count: 8
+        :assigned_system_count: 10
+        :unsupported_system_count: 2
+        :account: ${account}
+  :query: (percent_compliant = 20)
+  :except_parents:
+    - :systems
+- :name: "greater or equal search by percent_compliant"
```
Comment (RoamingNoMaD) [inline spec/fixtures/files/searchable/reports_controller.yaml:176]: This should be `greater search by percent_compliant`

**Review by RoamingNoMaD — CHANGES_REQUESTED** [review]
One slight naming issue, otherwise good work!

**Comment by romanblanco** [comment]
Great work, @LightOfHeaven1994!

This will also need to update OpenAPI spec:

```
docker-compose exec rails bundle exec rake rswag:specs:swaggerize
```

File: spec/fixtures/files/searchable/reports_controller.yaml (hunk 2 above)
Comment (LightOfHeaven1994) [reply spec/fixtures/files/searchable/reports_controller.yaml]: @RoamingNoMaD right! thanks, fixed

**Comment by LightOfHeaven1994** [comment]
> Great work, @LightOfHeaven1994!
> 
> This will also need to update OpenAPI spec:
> 
> ```
> docker-compose exec rails bundle exec rake rswag:specs:swaggerize
> ```

new spec is generated by running `docker compose exec rails bundle exec rake rswag:specs:swaggerize`

**Review by RoamingNoMaD — APPROVED** [review]
lgtm!

**Pull request merged**